lib/services/data_reader.dart

```diff
@@ -1,39 +1,76 @@
 import 'package:cloud_firestore/cloud_firestore.dart';
-import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
+import 'package:todo_app/constants/spaces.dart';
+import 'package:todo_app/themes/colors.dart';
 
+import '../constants/assets.dart';
 import '../screens/home/widget/task_tile.dart';
 
 class DataReader extends StatelessWidget {
-  const DataReader({super.key});
+  const DataReader({Key? key}) : super(key: key);
 
   @override
   Widget build(BuildContext context) {
     return StreamBuilder<QuerySnapshot>(
       stream: FirebaseFirestore.instance.collection('todos').snapshots(),
       builder: (BuildContext context, AsyncSnapshot<QuerySnapshot> snapshot) {
-        if (snapshot.hasData) {
-          final List<FiAs3AmnaTaskTile> taskTiles = snapshot.data!.docs.map(
-            (doc) {
-              final String taskText = doc['taskContent'];
-              final bool hasCompleted = doc['hasCompleted'];
+        try {
+          if (snapshot.hasData) {
+            final List<TaskTile> taskTiles = snapshot.data!.docs
```

Comment:
instead of checking this list to be empty later on just check above if snapshot.data.docs instead